opencart 2.2.0.0/upload/catalog/model/smailyforopencart/helper.php

```diff
@@ -78,8 +78,14 @@ public function apiCall($endpoint, array $data = [], $method = 'GET') {
      * @return array $customers All subscribed customers in array.
      */
     public function getSubscribedCustomers($offset) {
+        $this->load->model('setting/setting');
+        $sync_time = $this->model_setting_setting->getSettingValue('module_smaily_for_opencart_sync_time');
```

Comment:
This is a bad practice to fetch last sync time on each time you are fetching a chunk of customers from the database for 2 reasons:
- last synchronization time is not meant to change during CRON job run, so you are executing needless database queries;
- if last synchronization time should change then customers fetch will start to return different results and that can lead to odd synchronization behaviour;

A better approach is to load data before customers are fetched and provide the date as a `getSubscribedCustomers` function argument.